core/libp2p/transport/impl/transport_parser.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/transport/impl/transport_parser.hpp"
+
+#include <boost/asio/ip/address.hpp>
+#include <boost/lexical_cast.hpp>
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::transport, TransportParser::Error, e) {
+  using E = libp2p::transport::TransportParser::Error;
+  switch (e) {
+    case E::PROTOCOLS_UNSUPPORTED:
+      return "This protocol is not supported by libp2p transport";
+  }
+  return "Unknown error";
+}
+
+namespace libp2p::transport {
+
+  using boost::lexical_cast;
+  using boost::asio::ip::make_address;
+  using multi::Protocol;
+
+  // clang-format off
+  const std::list<std::list<Protocol::Code>> TransportParser::kSupportedProtocols {
```

Comment:
Is there a possibility that there would be more than two elements in a nested list? If not, better be pair.
Also, why lists instead of vectors?

Reply:
Use vector of vectors